pennylane/transforms/compile.py

```diff
@@ -133,6 +138,7 @@ def qfunc(x, y, z):
             if not isinstance(p_func, single_tape_transform) and not hasattr(p_func, "tape_fn"):
                 raise ValueError("Invalid transform function {p} passed to compile.")
 
+    pipeline.append(remove_barrier)
```

Comment:
Why not add `remove_barrier` to `default_pipeline`?

Then, if someone passes their own pipeline, they aren't removing the barrier before they explicitly want to. This will especially be useful if someone wants to do multiple compilation passes.